examples/step-6/doc/results.dox

```diff
@@ -484,6 +484,36 @@ step-49, step-53, and step-54, among other tutorial programs that cover
 the issue. Information on curved domains can also be found in the
 documentation module on @ref manifold "Manifold descriptions".
 
+The reason to track the geometry is due to the fact, that the finite
+element method uses quadrature integration over the cell.  This way,
+during the assembly of a local matrix all functions for the coefficient
+with same values at quadrature points will contribute to the integral
+exactly as a defined by a <code>quadrature_formula</code> polynomial
+fitted to these points. The mesh used here and in the step-5 is not
+aligned to the internal boundary, so it is possible to have both
+coefficient values of 1 or 20 at quadrature points inside of a single
+cell. This means that the abrupt change of the coefficient will be
+effectively replaced with a smooth polynomial. Moreover, as soon as
+the quadrature rule used in both steps does not have points on cell`s
+edges\faces this will most likely lead to the discontinuity of the
+polynomial representation (as it is used in the integration) of the
+coefficient on the boundary of such a cell. An important point here is
+that the discontinuity due to quadrature integration does not lead to
+the discontinuity of the solution (which is for sure is continuous due
+to the fact that it is what we obtain during the solution of the
+global system matrix using continuous finite elements).
+
+All this details are not important if we have a fixed set of materials
+and the interface between them is exactly on the boundary between
+cells. Refer to CellAccessor::set_material_id and step-28 as an
+advanced usage example.
+
+Finally, let us consider the case of a coefficient with non-uniform
+distribution in space. We can repeat once again all the discussion on
+the representation of such a function with quadratures. So, to
+simulate it correctly you have basically two options: you can reduce
+the cell size and/or increase the order of the polynomial used in the
+quadrature formula.
```

Comment:
I'm sorry if I seem overly critical, but I've read this a couple of times and I don't think that what you've said here is entirely correct or that its particularly easy to understand. There are also some more subtle points that the first paragraph does not accommodate, e.g. with XFEM and an appropriate choice of enrichment function (+ level-set function) one can capture this discontinuity exactly (or with higher precision if one wanted). Having browsed over the documentation for step-6, I now see that the next section ("Playing with the regularity of the solution") actually has a discussion on the influence of the jump in material coefficient on the solution. I think that this in some sense covers what you've tried to convey in the first paragraph (although its not discussing the choice of basis functions, which I've come to think is unnecessarily involved given the surrounding discussion).

I don't have any specific suggestions on how what you've written can be amended without completely revising it, so I will wait for @bangerth's  suggestions on how we should proceed. I am tempted to suggest that one need only focus on the following question: How can one do better than the second mesh presented in "A better mesh", and what benefits does it bring? For this, here are my cliff notes:

- One can change the mesh so that the line of the material discontinuity is resolved (in a discrete sense) along a set of interior cell faces.
- This means that on either side of the cell face, all points are considered to have a certain material coefficient. In this example this implies that the material coefficient is constant on a cell, although in general that need not be the case (as you'd inferred in the last paragraph, if the coefficient varies smoothly in space and the transition is "sufficiently gradual" then through refinement we can still render an accurate solution). 
- Through adjustment of the geometry and other assumptions we now have a genuine material interface as opposed to [an interphase](http://onlinelibrary.wiley.com/doi/10.1002/adma.19970090614/abstract). We have thus effectively divided the domain into two distinct material zones that do not overlap in the interior of a finite element.
- Since the material discontinuity leads to a jump in the solution gradient, this new discretisation that has support points along this line of discontinuity is able to capture this with enhanced accuracy. With the chosen finite element the solution, of course, remains continuous at this material interface. This better reflects the characteristics of the solution, which is not globally C1 continuous (it has a kink). Within each element, due to the chosen basis functions, the solution is approximated as one that is C1 continuous, and when there is a sharp interphase this does not accurately resolve the solution (nor its first derivative).
- It is no longer reasonable to use the analytical nonconstant `coefficient` function to describe the now discrete approximation of the material constant. However, by setting a cell `material_id` one can query the material coefficient based on the pre-defined material zone.
- [For some problems this approach is makes it much easier to resolve some physical phenomena and quantities, e.g. the normal jump in the magnetic field across materials, or the determination of the tractions at material interfaces such as the ponderomotive traction in electro-magnetics, or pressure loads for fluid-structure interaction).]